internal/gitdb/http.go

```diff
@@ -137,6 +137,8 @@ func (h *CheckoutHandler) SetupPublicJWTHandler(muxRouter *mux.Router, keyFunc j
 	muxRouter.Methods(http.MethodGet).Path("/public/file/{repo}/{branch}/{path:.*}").Handler(publicRepoMiddleware(middleware.Handler(httpserver.BasicHandler(h.getFileHandler, h.Log)))).Name("public_get_file_handler")
 	muxRouter.Methods(http.MethodGet).Path("/public/ls/{repo}/{branch}/{dir:.*}").Handler(publicRepoMiddleware(middleware.Handler(httpserver.BasicHandler(h.lsDirHandler, h.Log)))).Name("public_ls_dir_handler")
 	muxRouter.Methods(http.MethodGet).Path("/public/zip/{repo}/{branch}/{dir:.*}").Handler(publicRepoMiddleware(middleware.Handler(httpserver.BasicHandler(h.zipDirHandler, h.Log)))).Name("public_zip_dir_handler")
+	muxRouter.Methods(http.MethodGet).Path("/refresh/{repo}").Handler(publicRepoMiddleware(middleware.Handler(httpserver.BasicHandler(h.refreshRepoHandler, h.Log)))).Name("refresh_repo")
+	muxRouter.Methods(http.MethodGet).Path("/refreshall").Handler(middleware.Handler(httpserver.BasicHandler(h.refreshAllRepoHandler, h.Log)))).Name("refresh_all")
```

Comment:
This change likely requires significant changes, as I don't full understand the following:
1. what is the intended way to get gitdb to do a `git fetch`?
2. why were the refresh endpoints not exposed here?
3. why the refresh endpoints are HTTP POST methods?

Reply:
1. what is the intended way to get gitdb to do a git fetch?

When github sends a webhook that a push was done

2., why were the refresh endpoints not exposed here?

Not needed.  Github webhooks do refresh

3. why the refresh endpoints are HTTP POST methods?

They are modification requests, so they are POST.  Read only requests are GET